lib/msal-browser/src/utils/BrowserProtocolUtils.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { InteractionType } from "./BrowserConstants";
+import { StringUtils, ClientAuthError, ICrypto } from "@azure/msal-common";
+
+export type BrowserStateObject = {
+    interactionType: InteractionType
+};
+
+export class BrowserProtocolUtils {
+
+    /**
+     * Generates the state value used by the platform library
+     * @param browserCrypto 
+     * @param requestInteractionType 
+     */
+    static generateBrowserRequestState(browserCrypto: ICrypto, requestInteractionType: InteractionType): string {
+        if (StringUtils.isEmpty(requestInteractionType)) {
+            return "";
+        }
+
+        const stateObj: BrowserStateObject = {
+            interactionType: requestInteractionType
+        };
+
+        const stateString = JSON.stringify(stateObj);
+
+        return browserCrypto.base64Encode(stateString);
```

Comment:
I don't think this needs to be encoded here. What about returning the `stateObj` and then adding that as is to the library state object? That way we are only encoding/stringifying/decoding/parsing once.

Reply:
I didn't want common to specify the exact object structure, which is why I made it a string. I felt like adding `any` type for platformState was too generic. We can talk about this offline.